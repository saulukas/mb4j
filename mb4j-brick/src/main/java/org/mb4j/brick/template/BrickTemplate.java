package org.mb4j.brick.template;

import com.samskivert.mustache.Template;

public class BrickTemplate {

    public final String name;
    public final Template template;

    public BrickTemplate(String name, Template template) {
        this.name = name;
        this.template = template;
    }
}
