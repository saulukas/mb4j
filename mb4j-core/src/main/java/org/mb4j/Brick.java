package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Mustache.Lambda;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class Brick implements Lambda {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderIndented(this, frag);
  }
}
