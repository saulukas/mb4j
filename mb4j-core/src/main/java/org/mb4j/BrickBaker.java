package org.mb4j;

public interface BrickBaker<P> {
  Brick bakeBrick(P params);
}
