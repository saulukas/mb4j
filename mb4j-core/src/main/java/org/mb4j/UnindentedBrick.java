package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class UnindentedBrick extends Brick {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderUnindented(this, frag);
  }
}
