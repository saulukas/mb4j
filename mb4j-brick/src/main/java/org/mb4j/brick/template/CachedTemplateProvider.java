package org.mb4j.brick.template;

import java.util.HashMap;
import java.util.Map;
import org.mb4j.brick.Brick;

public class CachedTemplateProvider implements TemplateProvider {
  private final TemplateProvider source;
  private final Map<Class<? extends Brick>, BrickTemplate> class2template = new HashMap<>();

  public CachedTemplateProvider(TemplateProvider source) {
    this.source = source;
  }

  @Override
  public synchronized BrickTemplate templateFor(Class<? extends Brick> brickClass) {
    BrickTemplate result = class2template.get(brickClass);
    if (result == null) {
      result = source.templateFor(brickClass);
      class2template.put(brickClass, result);
    }
    return result;
  }
}
