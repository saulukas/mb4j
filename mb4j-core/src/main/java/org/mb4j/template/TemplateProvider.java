package org.mb4j.template;

import org.mb4j.Brick;

public interface TemplateProvider {
  BrickTemplate templateFor(Class<? extends Brick> brickClass);
}
