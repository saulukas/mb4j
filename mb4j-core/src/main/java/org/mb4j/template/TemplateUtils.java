package org.mb4j.template;

import org.mb4j.Brick;

public class TemplateUtils {
  public static String typeStringOf(Class<? extends Brick> brickClass) {
    TemplateType annotation = brickClass.getAnnotation(TemplateType.class);
    return annotation == null ? TemplateType.DEFAULT : annotation.value();
  }

  public static String encodingStringOf(Class<? extends Brick> brickClass) {
    TemplateEncoding annotation = brickClass.getAnnotation(TemplateEncoding.class);
    return annotation == null ? TemplateEncoding.DEFAULT : annotation.value();
  }

  public static String outputEncodingStringOf(Class<? extends Brick> brickClass) {
    TemplateOutputEncoding annotation = brickClass.getAnnotation(TemplateOutputEncoding.class);
    return annotation == null ? TemplateOutputEncoding.DEFAULT : annotation.value();
  }
}
