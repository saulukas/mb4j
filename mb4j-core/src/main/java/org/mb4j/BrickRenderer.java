package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import org.mb4j.template.TemplateProvider;
import java.io.Writer;

public class BrickRenderer {
  private final TemplateProvider templateProvider;

  public BrickRenderer(TemplateProvider templateProvider) {
    this.templateProvider = templateProvider;
  }

  public void render(Brick brick, Writer out) {
    RenderingScope scope = new RenderingScope(out, templateProvider);
    scope.render(brick);
  }
}
