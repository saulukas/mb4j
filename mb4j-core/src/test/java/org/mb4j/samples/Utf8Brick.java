package org.mb4j.samples;

import org.mb4j.Brick;
import org.mb4j.template.TemplateType;

@TemplateType(".mustache")
public class Utf8Brick extends Brick {
  final String ltCows = "žalos karvės ežere";
}
