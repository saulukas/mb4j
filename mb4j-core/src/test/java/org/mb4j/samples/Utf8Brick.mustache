'Brown cows in a lake' in lithuanian is '{{ltCows}}'.