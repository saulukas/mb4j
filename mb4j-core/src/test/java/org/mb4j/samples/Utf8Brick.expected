'Brown cows in a lake' in lithuanian is 'žalos karvės ežere'.